bin/rust/rust_belt/lifetime_logic.gh

```diff
@@ -78,6 +102,21 @@ lemma void close_full_borrow(predicate() P);
     requires close_full_borrow_token(P, ?q, ?k) &*& P();
     ensures full_borrow(k, P) &*& [q]lifetime_token(k);
 
+// LftL-bor-acc-strong
+predicate close_full_borrow_token_strong(lifetime_t k1, predicate() P, real q, lifetime_t k);
+
+lemma void open_full_borrow_strong(lifetime_t k, predicate() P, real q);
+    requires full_borrow(k, P) &*& [q]lifetime_token(k);
+    ensures exists(?k1) &*& lifetime_inclusion(k, k1) == true &*& P() &*& close_full_borrow_token_strong(k1, P, q, k);
+
+typedef lemma void full_borrow_convert_strong(predicate() Q, lifetime_t k1, predicate() P)();
```

Comment:
Add a TODO saying that in RustBelt this viewshift is at mask Bot so certain lemmas must not be called to prove this lemma. (For example, RustBelt does not support converting `[_]lifetime_dead_token(k1)` to `[_]lifetime_dead_token(k)` inside this proof.)

Reply:
One way to enforce this would be to introduce a chunk `current_mask(mask_t)` which is in the pre and post of *all* Rust functions and all lemmas. Another way is to have two versions of each lemma: one that is `nonghost_callers_only`, and one that consumes and produces a `current_mask`.